api/v1alpha1/catalogentry_types.go

```diff
@@ -0,0 +1,108 @@
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	kcpv1alpha1 "github.com/kcp-dev/kcp/pkg/apis/apis/v1alpha1"
+	conditionsv1alpha1 "github.com/kcp-dev/kcp/pkg/apis/third_party/conditions/apis/conditions/v1alpha1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	// InternalAPIBindingEntryLabelKey is the label key on an APIBinding with the
+	// base62(sha224(<workspace>:<exportName>)) as value to filter bindings by catalog entry.
+	InternalAPIBindingEntryLabelKey = "internal.catalog.kcp.dev/entry"
+)
+
+// These are valid conditions of CatalogEntry.
+const (
+	// CatalogEntryValid is a condition for CatalogEntry that reflects the validity
+	// of the referenced APIExport.
+	CatalogEntryValid conditionsv1alpha1.ConditionType = "CatalogEntryValid"
+	// CatalogEntryInvalidReferenceReason is a reason for the CatalogEntryValid
+	// condition of APIBinding that the referenced CatalogEntry reference is invalid.
+	CatalogEntryInvalidReferenceReason = "CatalogEntryInvalidReference"
+)
+
+//+kubebuilder:object:root=true
+//+kubebuilder:subresource:status
+
+// CatalogEntry is the Schema for the catalogentries API
+type CatalogEntry struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   CatalogEntrySpec   `json:"spec,omitempty"`
+	Status CatalogEntryStatus `json:"status,omitempty"`
+}
+
+// CatalogEntrySpec defines the desired state of CatalogEntry
+type CatalogEntrySpec struct {
+	// references is a list of references to APIExports.
+	References []kcpv1alpha1.ExportReference `json:"references"`
+	// icon is the representation data that can be displayed in UI
+	// +optional
+	Icon Icon `json:"icon,omitempty"`
+	// description is a human-readable message to describe the information regarding
+	// the capabilities and features that the API provides
+	// +optional
+	Description string `json:"description,omitempty"`
+}
+
+// Icon is the graphical representation that can be displayed in UI component
+type Icon struct {
+	// data is the encoded icon data
+	Data []byte `json:"data"`
+	// mediaType is media type information for icon data
+	MediaType string `json:"mediaType"`
+}
+
+// CatalogEntryStatus defines the observed state of CatalogEntry
+type CatalogEntryStatus struct {
+	// permissionClaims is a list of permissions requested by the API provider(s)
+	// for this catalog entry.
+	// +optional
+	PermissionClaims []kcpv1alpha1.PermissionClaim `json:"PermissionClaims,omitempty"`
+	// apiResources is the list of APIs that are provided by this catalog entry.
+	// +optional
+	APIResources []metav1.GroupResource `json:"apiResources,omitempty"`
```

Comment:
elsewhere we just write `resources`. Would do it here too. What else should be than apiResources?